app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,354 @@
+    }
+
+    /**
+     * Use this function to modify the daily distance goal of the user
+     * @param distance new daily distance goal
+     */
+    fun setDistanceGoal(distance: Double){
+        if(distance == 0.0){
+            throw java.lang.Error("The distance goal can't be equal to 0 !")
+        }
+        this.distanceGoal=distance
+    }
+
+    /**
+     * Get the daily activity time goal of the user
+     * @return daily activity time goal of the user
+     */
+    fun getActivityTime(): Double{
+        return activityTimeGoal
+    }
+
+    /**
+     * Use this function to modify the daily activity time goal of the user
+     * @param time new daily activity time goal
+     */
+    fun setActivityTimeGoal(time: Double){
+        if(time == 0.0){
+            throw java.lang.Error("The activity time goal can't be equal to 0 !")
+        }
+        this.activityTimeGoal=time
+    }
+
+    /**
+     * Get the daily number of paths goal of the user
+     * @return daily number of paths goal of the user
+     */
+    fun getNumberOfPathsGoal(): Int{
+        return nbOfPathsGoal
+    }
+
+    /**
+     * Use this function to modify the daily number of paths goal of the user
+     * @param nbOfPaths new daily number of paths goal
+     */
+    fun setNumberOfPathsGoal(nbOfPaths: Int){
+        if(nbOfPaths == 0){
+            throw java.lang.Error("The number of paths goal can't be equal to 0 !")
+        }
+        this.nbOfPathsGoal=nbOfPaths
+    }
+
+    /**
+     * Get the age of the user
+     * @return the age of the user
+     */
+    fun getAge(): Int{
+        var age = LocalDate.now().year-dateOfBirth.year
+        if(LocalDate.now().minusYears(age.toLong())<=dateOfBirth){
+            age -= 1
+        }
+        return age
+    }
+
+    /**
+     * This function will remove the user with username to the friend list
+     * @param username of the user that we want to remove
+     */
+    fun removeFriend(username: String) {
+        if(!friendsList.contains(username)) {
+            throw java.lang.Error("This user is not in the friend list !")
+        }
+        friendsList.remove(username)
+    }
+
+    /**
+     * This function will add the user with username to the friend list. To be added the user must be present in the database.
+     * @param username of the user that we want to add to the friend list
+     */
+    fun addFriend(username: String) {
+        //TODO:this function will be implemented during a next task when the database will be cleaned
+    }
+
+    /**
+     * This function will return the friend list of a user
+     * @return the friend list of the user
+     */
+    fun getFriendList():Map<String, String> {
+        return this.friendsList
+    }
+
+}
+
+/**
+ * Helper function to check the userId and affect it if it's correct
+ * @param userId that correspond to the user
+ * @param database where the userId should be present
+ */
+private fun checkUserId(userId: String, database: Database):Boolean{
```

Comment:
Doing a synchronous check is not possible here.

Reply:
I have change it with a return of a CompletableFure<Boolean>